packages/suite/src/actions/settings/walletSettingsActions.ts

```diff
@@ -83,3 +87,43 @@ export const removeBlockbookUrl = (payload: BlockbookUrl) => ({
     type: WALLET_SETTINGS.REMOVE_BLOCKBOOK_URL,
     payload,
 });
+
+export const setTorBlockbookUrls = () => async (dispatch: Dispatch, getState: GetState) => {
+    const { blockbookUrls } = getState().wallet.settings;
+    const noTouchyCoins = blockbookUrls.filter(u => !u.tor).map(u => u.coin);
+    const coins = NETWORKS.filter(n => !n.accountType && !noTouchyCoins.includes(n.symbol)).map(
+        n => n.symbol,
+    );
+
+    const torBlockbook: BlockbookUrl[] = [];
+    // Disabling this ESLint rule here because it doesn't make sense for this scenario.
+    // It suggests using forEach which not waiting for the loop to finish before going
+    // to the next statement, resulting in `torBlockbook` to be empty.
+    // eslint-disable-next-line no-restricted-syntax
+    for await (const coin of coins) {
+        const resp = await TrezorConnect.getCoinInfo({ coin });
+        if (resp.success) {
+            if (resp.payload.blockchainLink?.type === 'blockbook') {
+                const urls = resp.payload.blockchainLink.url;
+                if (urls.find(u => u.endsWith('trezor.io') !== undefined)) {
+                    urls.forEach(u => {
+                        torBlockbook.push({
+                            coin,
+                            url: toTorUrl(u),
+                            tor: true,
+                        });
+                    });
+                }
+            }
+        }
+    }
```

Comment:
this could be also written as:
```
    const promises = coins.map(coin => TrezorConnect.getCoinInfo({ coin }));
    const results = await Promise.all(promises);
    results.forEach(resp => {
        if (resp.success && resp.payload.blockchainLink?.type === 'blockbook') {
            // ... set tor urls
        }
    }
```

Reply:
Good point, would also help getting rid of the eslint exception. Will update, thanks!

Reply:
1e0f157